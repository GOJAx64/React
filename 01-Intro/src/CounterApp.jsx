import PropTypes from 'prop-types';
import { useState } from 'react';

export const CounterApp = ( {value} ) => {
    const [counter, setCounter] = useState(value);
    
    const addValue = () => setCounter(counter + 1);
    //We can do it different, try ->  setCounter( (c) => c + 1 )

    const subsValue = () => setCounter(counter - 1);

    const resetValue = () => setCounter(value);

    return (
        <>
            <h1>CounterApp</h1>
            <h2> { counter } </h2> 
            <button onClick={ addValue }> +1 </button>  
            <button onClick={ subsValue }> -1 </button>
            <button onClick={ resetValue }> Reset </button>
        </>
  )
}


CounterApp.propTypes = {
    value: PropTypes.number.isRequired,
}

CounterApp.defaultProps = {
    value: 0,
}