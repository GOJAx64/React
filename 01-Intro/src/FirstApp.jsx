import PropTypes from 'prop-types';

//We can use fragmet, but it's not used a lot
const message = {
    age: 22,
    me: 'Alberto', 
}

const printMessage = (message) => message.age + ' ' + message.me

export const FirstApp = ( {title, number, subtitle} ) => {
  return (
    <>
        <h1>{ title }</h1>
        {/*<h1> { printMessage(message) } </h1>
        <p>This is a subtitle</p>
        <br/>*/}
        <p>{ subtitle }</p>
        <p>{ number }</p>
    </>
  )
}

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    number: PropTypes.number.isRequired,
}

FirstApp.defaultProps = {
    //title: 'Alberto Garc√≠a',
    number: 888,
    //subtitle: 'This is a default propertie',
}
