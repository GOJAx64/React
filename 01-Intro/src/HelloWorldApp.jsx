/*export function App() { //My first component
    return (<h1>Hello World</h1>)
}*/

//We can use this
//export default App;
//use rafc
export const HelloWorldApp = () =>  {
    return (
        <p>HelloWorld App</p>
    )
}
