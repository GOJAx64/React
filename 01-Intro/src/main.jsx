import React from 'react'
import ReactDOM from 'react-dom/client'
import {HelloWorldApp} from './HelloWorldApp'
import { FirstApp } from './FirstApp'
import './styles.css'
import { CounterApp } from './CounterApp'

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    {/*<HelloWorldApp/>
    <FirstApp title='title props' number={ 123 }/> */}
    <CounterApp value={0}/>
  </React.StrictMode>
)
