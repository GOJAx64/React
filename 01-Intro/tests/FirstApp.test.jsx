import { render } from "@testing-library/react";
import { FirstApp } from "../src/FirstApp";

describe('Tests in <FirstApp/>', () => { 
    
    test('Should do match with the snapshot', () => {
        const { container } = render(<FirstApp title={ 'Hola soy goku' } />);
        expect(container).toMatchSnapshot();
    });

    test('should show the title in  one H1', () => {  
        const title = 'Hola soy goku';
        const { container, getByText } = render(<FirstApp title={ title } />);
        expect(getByText(title)).toBeTruthy();

        const h1 = container.querySelector('h1');
        expect(h1.innerHTML).toContain(title);
    });
});