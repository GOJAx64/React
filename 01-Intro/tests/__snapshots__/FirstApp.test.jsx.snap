// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <FirstApp/> Should do match with the snapshot 1`] = `
<div>
  <h1>
    Hola soy goku
  </h1>
  <h1>
     
    22 Alberto
     
  </h1>
  <p>
    This is a subtitle
  </p>
  <br />
  <p>
    This is a 
    Hola soy goku
     and this is a number 
    888
     passing by props
  </p>
  <p>
    This is a default propertie
  </p>
</div>
`;
